tools/runners/parallels.sh

```diff
@@ -0,0 +1,410 @@
+      cp $_testfile $_queue
+    else
+      log "Test target extention can not be recognized: '$_testfile'"
+      exit 1
+    fi
+  elif [ -d $_testfile ]; then
+    find $_testfile -iname "[0-9]*.lua" | sort > $_queue
+    if [ ! -s $_queue ]; then
+      rm $_queue
+      log "Failed to find any test scripts in: '$_testfile'"
+      exit 1
+    fi
+  else
+    log "Unsupported test target format: '$_testfile'"
+    exit 1
+  fi
+
+  sed -E '/^;($|[^.])/d' -i $_queue
+  cp $_queue $_queue_reference
+}
+
+function common {
+  # remove tmp dir if exists
+  if [ -d "$_tmp_dir" ]; then
+    rm -r "$_tmp_dir"
+  fi
+  rm -f "$_test_id_file"
+
+  mkdir $_tmp_dir
+
+  prepare_queue
+  prepare_num_of_workers
+
+  prepare_sdl
+  prepare_atf
+}
+
+function mktempdir {
+  #
+  #   Create temporary folder with SDL, ATF and Test Scripts
+  #
+
+  tmpdirname=$(mktemp --suffix=_worker --tmpdir=$_tmp_dir -d)
+
+  mkdir -p $tmpdirname/bin
+
+  cp $_sdl_prepared/bin/hmi_capabilities.json $tmpdirname/bin/
+  cp $_sdl_prepared/bin/log4cxx.properties $tmpdirname/bin/
+  cp $_sdl_prepared/bin/*.pem $tmpdirname/bin/
+  cp $_sdl_prepared/bin/sdl_preloaded_pt.json $tmpdirname/bin/
+  cp $_sdl_prepared/bin/smartDeviceLink.ini $tmpdirname/bin/
+
+  ln -s ../../sdl_ext/bin/audio.8bit.wav $tmpdirname/bin/audio.8bit.wav
+  ln -s ../../sdl_ext/bin/build_config.txt $tmpdirname/bin/build_config.txt
+  ln -s ../../sdl_ext/bin/smartDeviceLinkCore $tmpdirname/bin/smartDeviceLinkCore
+
+  for f in $(find $_sdl_prepared/bin/ -name "lib*"); do
+    ln -s ../../sdl_ext/bin/$(basename $f) $tmpdirname/bin/
+  done
+
+  cp -r $atf_tmp_dir $tmpdirname/
+
+  if [ "$_copy_ts" = true ]; then
+    cp -r $atf_tmp_ts_dir/test_scripts $tmpdirname/$(basename $atf_tmp_dir)/
+    cp -r $atf_tmp_ts_dir/test_sets $tmpdirname/$(basename $atf_tmp_dir)/
+    cp -r $atf_tmp_ts_dir/user_modules $tmpdirname/$(basename $atf_tmp_dir)/
+    cp -r $atf_tmp_ts_dir/files $tmpdirname/$(basename $atf_tmp_dir)/
+  else
+    ln -s ../../atf_ts/test_scripts $tmpdirname/$(basename $atf_tmp_dir)
+    ln -s ../../atf_ts/test_sets $tmpdirname/$(basename $atf_tmp_dir)
+    ln -s ../../atf_ts/user_modules $tmpdirname/$(basename $atf_tmp_dir)
+    ln -s ../../atf_ts/files $tmpdirname/$(basename $atf_tmp_dir)
+  fi
+
+  echo $tmpdirname
+}
+
+function wait_screen_termination {
+  while screen -ls | grep -q ${_tmp_workers// /\\|}
+  do
+    sleep 1
+  done
+}
+
+function wait_screen_termination_with_progress {
+  while screen -ls | grep -q ${_tmp_workers// /\\|}
+  do
+    sleep 1
+    show_progress
+  done
+}
+
+function show_progress {
+  new_state=$(diff -N <(sed -e '$a\' $_queue) <(sed -e '$a\' $_queue_reference) | tail -n +2 | awk {'print $2'})
+  processed=$(diff <(echo "$new_state") <(echo "$_last_state") | tail -n +2 | awk '{print $2}')
+
+  if [ -z "$processed" ]; then
+    return
+  fi
+
+  for item in $processed
+  do
+    log $item
+  done
+  _last_state=$new_state
+}
+
+function clean_up {
+  rm -r $_tmp_dir
```

Comment:
```suggestion
  rm -rf $_tmp_dir
```
Seems that the bson_c_lib can be copied here sometimes, so I'm getting "rm: remove write-protected regular file" prompts during this step

Reply:
Probably several other places where this change would be needed as well.